mindsdb/integrations/libs/base.py

```diff
@@ -110,7 +110,75 @@ def __init__(self, name: str):
 
 class PredictiveHandler(BaseHandler):
     """
+    DEPRECATED.
+
     Base class for handlers associated to predictive systems.
     """
     def __init__(self, name: str):
         super().__init__(name)
+
+
+class BaseMLEngine:
+    """
+    Base class for integration engine to connect with other Machine Learning libraries/frameworks.
+
+    This class will be instanced when interacting with the underlying framework.
+    """
+
+    def __init__(self, model_storage, engine_storage) -> None:
+        """
+        Warning: This method should not be overridden.
+
+        Initialize storage objects required by the ML engine.
+
+        - engine_storage: persists global engine-related internals.
+        - model_storage: stores artifacts for any given model.
+        """
+        self.model_storage = model_storage
+        self.engine_storage = engine_storage
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        """
+        Saves a model inside the engine registry for later usage.
+
+        Normally, an input dataframe is required to train the model.
+        However, some integrations may merely require registering the model instead of training, in which case `df` can be omitted.
+
+        Any other arguments required to register the model can be passed in an `args` dictionary.
+        """
+        raise NotImplementedError
+
+    def predict(self, df: pd.DataFrame, args: Optional[Dict] = None) -> pd.DataFrame:
+        """
+        Calls a model with some input dataframe `df`, and optionally some arguments `args` that may modify the model behavior.
+
+        The expected output is a dataframe with the predicted values in the target-named column.
+        Additional columns can be present, and will be considered row-wise explanations if their names finish with `_explain`.
+        """
+        raise NotImplementedError
+
+    def update(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        """
+        Optional.
+
+        Used to update/fine-tune/adjust a pre-existing model without resetting its internal state (e.g. weights).
+
+        Availability will depend on underlying integration support, as not all ML models can be partially updated.
+        """
+        raise NotImplementedError
+
+    def describe(self, key: Optional[str] = None) -> pd.DataFrame:
```

Comment:
`HandlerResponse` instead of pd.DataFrame

Reply:
Done by the wrapper `libs/ml_exec_base.py:BaseMLEngineExec`.